bindings/rust/s2n-tls/src/testing/resumption.rs

```diff
@@ -0,0 +1,335 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#[cfg(test)]
+mod tests {
+    use crate::{
+        connection,
+        session_ticket::{SessionTicket, SessionTicketCallback},
+        testing::{s2n_tls::*, *},
+    };
+    use std::{cell::RefCell, rc::Rc, time::Duration};
+
+    // Creates session ticket callback handler
+    #[derive(Default, Clone)]
+    pub struct SessionTicketRawBytesHandler {
+        stored_ticket: Rc<RefCell<Vec<u8>>>,
+    }
+
+    // Implement the session ticket callback
+    impl SessionTicketCallback for SessionTicketRawBytesHandler {
```

Comment:
I'm not entirely certain why having two different test SessionTicketCallbacks is necessary? Could the "default ticket lifetime" assert just be implemented in the main `SessionTicketHandler`?

Reply:
That is actually a red herring for why we need the second callback test, and probably my fault for being confusing. It's trying to test if the client had stored only the session ticket bytes and needed to create a new SessionTicket in order to call set_session.